docs/content/docs/try-flink/write_flink_program_with_sql.md

```diff
@@ -0,0 +1,269 @@
+
+
+## Step 4: Run your first continuous SQL query and learn about dynamic tables
+
+Now use the DQL syntax `SELECT` to view the streaming data in this table:
+
+```sql
+SELECT * FROM twitch_stream;
+```
+
+You should see, on the console of the SQL client, a stream of data populating each of the defined fields of the twitch_stream table.  Notice that this table contains a stream of data and is what is known as a [dynamic table]({{< ref "docs/dev/table/concepts/dynamic_tables" >}}).  
+
+Dynamic tables are the fundamental concept behind Flink's SQL support for streaming data. While SQL makes it seem like you are querying a database (or static tables that represent batch data, the tables in Flink are actually dynamic tables that are defined by queries. So instead of running several different queries on the same set of data, you are continuously running one query on a dataset that keeps changing. 
+
+Under the hood, the SQL client submits queries to Flink's JobManager, which works with the TaskManager(s) to assign and monitor query tasks. Have a look at [Flink's architecture]({{< ref "docs/concepts/flink-architecture" >}}) for more detail. 
+
+
+## Step 5: Filter the data
+
+Now try to perform a filter operation on this data stream to specify a subset of the data by using the `WHERE` keyword.  
+
+To find out what gaming streams started in the last 15 minutes, try this query:
+
+```sql
+SELECT game_name, 
+       started_at,
+       proctime, 
+       TIMESTAMPDIFF(MINUTE, started_at, proctime)
+FROM twitch_stream
+WHERE TIMESTAMPDIFF(MINUTE, started_at, proctime) < 15;
+```
+
+You should now see a new table with new datastream results.  
+
+
+Now try a [Top-N]({{< ref "docs/dev/table/sql/queries/topn" >}}) query to find the 10 most popular games.
+
+Top-N queries identify the N smallest or largest values (as ordered by some attribute of the table), and are useful when you need to identify the top (or bottom) N items in a stream. 
+
+```sql
+SELECT game_name, proctime, viewer_count
+FROM (
+    SELECT *, ROW_NUMBER() OVER (ORDER BY viewer_count DESC) AS ranking
+    FROM (
+        SELECT *
+        FROM (
+            SELECT 
+                *,
+                ROW_NUMBER() OVER (PARTITION BY game_name ORDER BY proctime DESC) AS ordering
+            FROM twitch_stream
+        ) WHERE ordering = 1
+    )
+) WHERE ranking <= 10;
+```
+
+Note that Flink uses the combination of an OVER window clause and a filter condition to express a Top-N query in order to filter through unbounded dataset. 
+
+You should now see a new dynamic table with the results from this query.  Notice how the top 10 games are constantly being revised as new data is processed.
+
+
+## Step 6: Aggregate the data and learn about windowing 
+
+Now try an aggregate function (`COUNT`, `SUM`, `AVG`, `MIN`, `MAX`) to find the most popular games based on location with the following query: 
```

Comment:
This query isn't doing anything to limit the results to only the most popular games.